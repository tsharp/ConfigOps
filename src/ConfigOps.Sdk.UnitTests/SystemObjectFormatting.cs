using ConfigOps.Sdk.Dto;
using ConfigOps.Sdk.Serialization;
using NJsonSchema.Generation;
using NJsonSchema.NewtonsoftJson.Generation;
using Snapshooter.Xunit;

namespace ConfigOps.Sdk.UnitTests
{
    public class SystemObjectFormatting
    {
        [Fact]
        public void Validate_SystemObject_YamlFormat()
        {
            SystemObject systemObject = new SystemObject()
            {
                ApiVersion = "v1",
                Kind = "ServiceConfig",
                Metadata = new SystemObjectMetadata()
                {
                    Name = "service-a",
                    Namespace = "default",
                    Labels = new Dictionary<string, string>()
                    {
                        { "geo", "us" },
                        { "region", "westus" },
                        { "env", "prod" }
                    }
                }
            };

            systemObject
                .ObjectToYaml()
                .MatchSnapshot();
        }

        [Fact]
        public void GenerateSchemaWithNSwag()
        {
            var schema = JsonSchemaGenerator.FromType(typeof(SystemObject), new NewtonsoftJsonSchemaGeneratorSettings()
            {
                SchemaType = NJsonSchema.SchemaType.OpenApi3,
                FlattenInheritanceHierarchy = true,
                UseXmlDocumentation = true
            });

            var objectSchema = schema.Definitions
                .First()
                .Value;

            objectSchema.ToJson()
                .JsonToYaml()
                .MatchSnapshot();
        }
    }
}