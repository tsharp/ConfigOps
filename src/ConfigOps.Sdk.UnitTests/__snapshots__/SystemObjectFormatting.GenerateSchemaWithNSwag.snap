$schema: http://json-schema.org/draft-04/schema#
type: object
additionalProperties: {}
properties:
  name:
    type: string
  namespace:
    type: string
  labels:
    type: object
    x-nullable: true
    additionalProperties:
      type: string
  annotations:
    type: object
    x-nullable: true
    additionalProperties: {}
